nixos/modules/virtualisation/docker.nix

```diff
@@ -40,13 +40,24 @@ in
       };
     storageDriver =
       mkOption {
-        type = types.enum ["aufs" "btrfs" "devicemapper" "overlay" "zfs"];
-        default = "devicemapper";
+        type = types.nullOr (types.enum ["aufs" "btrfs" "devicemapper" "overlay" "zfs"]);
+        default = null;
         description =
           ''
             This option determines which Docker storage driver to use.
```

Comment:
Ok, will add in description. It's acually not new, we used to have by default to docker to choose driver, but it was changed later to default option that broke on some filesystems like btrfs, and was not a good choice for some others. Docker has a good priority list which device storage to use.